### PR TITLE
Add new DB2CoreDriver to use with IBM.Data.DB2.Core provider

### PR DESCRIPTION
**Pull request by hazzik**
7dfb32cadc5ce06fd8246b2dcf8b3150

### CODE REVIEW

File: src/NHibernate/Dialect/DB2Dialect.cs [hunk 1]
```diff
@@ -128,8 +114,6 @@ public DB2Dialect()
 
 			// DB2 does not support ANSI substring syntax.
 			RegisterFunction("substring", new SQLFunctionTemplate(NHibernateUtil.String, "substring(?1, ?2, ?3)"));
-
-			DefaultProperties[Environment.ConnectionDriver] = "NHibernate.Driver.DB2Driver";
```
Comment (fredericDelaporte) [inline src/NHibernate/Dialect/DB2Dialect.cs]: Removing this is a breaking change for all apps which were using the dialect without specifying the driver. Better keep-it, eventually guarded by a compilation directive for not specifying an invalid driver under .Net Core and a maybe invalid one under .Net Standard.

Comment (hazzik) [reply src/NHibernate/Dialect/DB2Dialect.cs]: Reverted

**Comment by fredericDelaporte** [comment]
HMM3. Your screenshot is from that game, of course a bit tweaked.

**Comment by hazzik** [comment]
Get it now.

**Pull request merged**